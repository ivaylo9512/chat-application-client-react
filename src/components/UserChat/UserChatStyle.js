import styled from 'styled-components'
import { ImageContainer } from '../Chat/ChatStyles'

export const Container = styled.div`
    height: 20vh;
    margin-bottom: 2vh;
    display: flex;
    justify-content: center;
    align-items: center;
`

export const InfoContainer = styled.div`
    display: inline-block;
    width: 19vw;
    height: 9.5vh;
    vertical-align: bottom;
    margin-left: 1vw;
    display: inline-flex;
    border-radius: 0.5vw;
    background: rgb(24,27,33);
    text-decoration: none;
    box-shadow: 0px 0px 0.15vw 0.15vw rgb(0 0 0);
    position: relative;
    display: flex;
    color: white;
`

export const Info = styled.div`
    display: flex;
    flex-direction: column;
    color: white;
    font-size: 1vw;
    flex: 1;
    text-align: center;
`

export const EnterButton = styled.button`
    background: rgb(24,27,33);
    margin-left: 1vw;
    height: 2.5vh;
    width: 2.5vh;
    font-size: 1vh;
    color: white;
    border-radius: 50%;
    box-shadow: 0px 0px 0.15vw 0.15vw rgb(0 0 0);
`

export const Image = styled(ImageContainer)`
    display: inline-flex;
`