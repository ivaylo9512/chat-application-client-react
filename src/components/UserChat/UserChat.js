import { useState } from 'react';
import { IMAGE_URL } from 'appConstants'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faForward, faInfo } from '@fortawesome/free-solid-svg-icons'
import { Container, Image, Info, ButtonsContainer, Button, InfoContainer } from './UserChatStyle';
import { useDispatch } from 'react-redux';
import { setCurrentChat } from 'app/slices/chatsSlice';

const UserChat = ({userChat, userChat: {createdAt, secondUser: { profileImage, lastName, firstName }}}) => {
    const dispatch = useDispatch();
    const [isInfoVisible, setIsInfoVisible] = useState(false);

    const setChat = () => {
        dispatch(setCurrentChat(userChat))
    }

    return(
        <Container>
            <Image className='image-container'>
                <img src={`${IMAGE_URL}/${profileImage}`}/>
            </Image>
            <InfoContainer isInfoVisible={isInfoVisible}>
                <span><b>{`${firstName} ${lastName}`}</b></span>
                <Info isInfoVisible={isInfoVisible}>
                   <span><b>{`${firstName} ${lastName}`}</b></span>
                   <span><b>{createdAt}</b></span>
                </Info>
            </InfoContainer>
            <ButtonsContainer>
                <Button data-testid='toggleInfo' onClick={() => setIsInfoVisible(!isInfoVisible)}><FontAwesomeIcon icon={faInfo}/></Button>
                <Button data-testid='setChat' onClick={setChat}><FontAwesomeIcon icon={faForward}/></Button>
            </ButtonsContainer>
        </Container>
    )
}
export default UserChat