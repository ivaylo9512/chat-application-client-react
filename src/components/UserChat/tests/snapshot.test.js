import * as redux from 'react-redux';
import { shallow } from 'enzyme';
import UserChat from 'components/UserChat/UserChat';

describe('UserChat snapshot tests', () => {
    let wrapper;

    beforeEach(() => {
        jest.spyOn(redux, 'useDispatch').mockReturnValue(jest.fn());

        wrapper = shallow(<UserChat userChat={{id: 1, createdAt:'2021-08-22', secondUser: { profileImage: 'image.png', firstName: 'firstname', lastName: 'lastName' }}}/>)
    })

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot with toggled info', () => {
        wrapper.findByTestid('toggleInfo').props().onClick();
        expect(wrapper).toMatchSnapshot();
    })
})