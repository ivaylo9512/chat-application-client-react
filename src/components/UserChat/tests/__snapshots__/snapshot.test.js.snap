// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserChat snapshot tests should match snapshot 1`] = `
<styled.div>
  <Styled(styled.button)
    className="image-container"
  >
    <img
      src="http://localhost:8080/images/download/image.png"
    />
  </Styled(styled.button)>
  <styled.div
    isInfoVisible={false}
  >
    <span>
      <b>
        firstname lastName
      </b>
    </span>
    <styled.div
      isInfoVisible={false}
    >
      <span>
        <b>
          firstname lastName
        </b>
      </span>
      <span>
        <b>
          2021-08-22
        </b>
      </span>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.button
      data-testid="toggleInfo"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              192,
              512,
              Array [],
              "f129",
              "M20 424.229h20V279.771H20c-11.046 0-20-8.954-20-20V212c0-11.046 8.954-20 20-20h112c11.046 0 20 8.954 20 20v212.229h20c11.046 0 20 8.954 20 20V492c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20v-47.771c0-11.046 8.954-20 20-20zM96 0C56.235 0 24 32.235 24 72s32.235 72 72 72 72-32.235 72-72S135.764 0 96 0z",
            ],
            "iconName": "info",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </styled.button>
    <styled.button
      data-testid="setChat"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f04e",
              "M500.5 231.4l-192-160C287.9 54.3 256 68.6 256 96v320c0 27.4 31.9 41.8 52.5 24.6l192-160c15.3-12.8 15.3-36.4 0-49.2zm-256 0l-192-160C31.9 54.3 0 68.6 0 96v320c0 27.4 31.9 41.8 52.5 24.6l192-160c15.3-12.8 15.3-36.4 0-49.2z",
            ],
            "iconName": "forward",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </styled.button>
  </styled.div>
</styled.div>
`;

exports[`UserChat snapshot tests should match snapshot with toggled info 1`] = `
<styled.div>
  <Styled(styled.button)
    className="image-container"
  >
    <img
      src="http://localhost:8080/images/download/image.png"
    />
  </Styled(styled.button)>
  <styled.div
    isInfoVisible={true}
  >
    <span>
      <b>
        firstname lastName
      </b>
    </span>
    <styled.div
      isInfoVisible={true}
    >
      <span>
        <b>
          firstname lastName
        </b>
      </span>
      <span>
        <b>
          2021-08-22
        </b>
      </span>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.button
      data-testid="toggleInfo"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              192,
              512,
              Array [],
              "f129",
              "M20 424.229h20V279.771H20c-11.046 0-20-8.954-20-20V212c0-11.046 8.954-20 20-20h112c11.046 0 20 8.954 20 20v212.229h20c11.046 0 20 8.954 20 20V492c0 11.046-8.954 20-20 20H20c-11.046 0-20-8.954-20-20v-47.771c0-11.046 8.954-20 20-20zM96 0C56.235 0 24 32.235 24 72s32.235 72 72 72 72-32.235 72-72S135.764 0 96 0z",
            ],
            "iconName": "info",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </styled.button>
    <styled.button
      data-testid="setChat"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f04e",
              "M500.5 231.4l-192-160C287.9 54.3 256 68.6 256 96v320c0 27.4 31.9 41.8 52.5 24.6l192-160c15.3-12.8 15.3-36.4 0-49.2zm-256 0l-192-160C31.9 54.3 0 68.6 0 96v320c0 27.4 31.9 41.8 52.5 24.6l192-160c15.3-12.8 15.3-36.4 0-49.2z",
            ],
            "iconName": "forward",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </styled.button>
  </styled.div>
</styled.div>
`;
